rtp/src/main/kotlin/org/jitsi/rtp/rtp/header_extensions/Av1DependencyDescriptorHeaderExtension.kt

```diff
@@ -0,0 +1,443 @@
+/*
+ * Copyright @ 2018 - present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.rtp.rtp.header_extensions
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings
+import org.jitsi.rtp.rtp.RtpPacket
+import org.jitsi.rtp.util.BitReader
+
+/**
+ * The subset of the fields of an AV1 Dependency Descriptor that can be parsed statelessly.
+ */
+open class Av1DependencyDescriptorStatelessSubset(
+    val startOfFrame: Boolean,
+    val endOfFrame: Boolean,
+    val frameDependencyTemplateId: Int,
+    val frameNumber: Int,
+
+    val newTemplateDependencyStructure: Av1TemplateDependencyStructure?,
+)
+
+/**
+ * The AV1 Dependency Descriptor header extension, as defined in https://aomediacodec.github.io/av1-rtp-spec/#appendix
+ */
+class Av1DependencyDescriptorHeaderExtension(
+    startOfFrame: Boolean,
+    endOfFrame: Boolean,
+    frameDependencyTemplateId: Int,
+    frameNumber: Int,
+
+    newTemplateDependencyStructure: Av1TemplateDependencyStructure?,
+
+    val activeDecodeTargetsBitmask: Int?,
+
+    val customDtis: List<DTI>?,
+    val customFdiffs: List<Int>?,
+    val customChains: List<Int>?,
+
+    val structure: Av1TemplateDependencyStructure
+) : Av1DependencyDescriptorStatelessSubset(
+    startOfFrame,
+    endOfFrame,
+    frameDependencyTemplateId,
+    frameNumber,
+    newTemplateDependencyStructure
+) {
+    val frameInfo: FrameInfo
+        get() {
```

Comment:
Why does this need to be recreated each time (as far as I see it just accesses `val`s)? Would a `lazy` delegate work instead?